concent_api/middleman/tests/test_middleman.py

```diff
@@ -0,0 +1,54 @@
+import os
+import signal
+import threading
+import time
+
+from assertpy import assert_that
+import mock
+import pytest
+
+from middleman.constants import EXIT_STATUS_ADDRESS_IN_USE_ERROR
+from middleman.constants import EXIT_STATUS_OK
+from middleman.middleman_server import MiddleMan
+
+
+class TestMiddleManServer:
+    patcher = None
+    crash_logger_mock = None
+
+    def setup_method(self):
+        self.patcher = mock.patch("middleman.middleman_server.crash_logger")
+        self.crash_logger_mock = self.patcher.start()
+
+    def teardown_method(self):
+        self.patcher.stop()
+
+    def test_that_if_ctrl_c_is_pressed_application_will_exit_with_status_0(self, event_loop):
+        with pytest.raises(SystemExit) as exception_wrapper:
+            middleman = MiddleMan(loop=event_loop)
+            with mock.patch.object(middleman, "_run_forever", side_effect=KeyboardInterrupt):
+                middleman.run()
+        assert_that(exception_wrapper.value.code).is_equal_to(EXIT_STATUS_OK)
+        self.crash_logger_mock.assert_not_called()
+
+    @mock.patch("middleman.middleman_server.asyncio.start_server", side_effect=OSError)
+    def test_that_if_chosen_port_is_already_used_application_will_exit_with_error_status(self,  _start_server_mock, event_loop):
+        with pytest.raises(SystemExit) as exception_wrapper:
+            MiddleMan(loop=event_loop).run()
+        assert_that(exception_wrapper.value.code).is_equal_to(EXIT_STATUS_ADDRESS_IN_USE_ERROR)
+        self.crash_logger_mock.assert_not_called()
+
+    def test_that_if_sigterm_is_sent_application_will_exit_with_status_0(self, event_loop):
+        pid = os.getpid()
+
+        def trigger_signal():
+            time.sleep(1)
+            os.kill(pid, signal.SIGINT)
+
+        thread = threading.Thread(target=trigger_signal)
```

Comment:
Isn't it actually killing current process ?

Reply:
no, test works and it's not interrupted.